service/src/test/java/bio/terra/profile/pact/provider/BPMProviderTest.java

```diff
@@ -0,0 +1,137 @@
+import au.com.dius.pact.provider.junitsupport.loader.PactBroker;
+import au.com.dius.pact.provider.spring.junit5.PactVerificationSpringProvider;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
+import bio.terra.profile.app.Main;
+import bio.terra.profile.app.configuration.BeanConfig;
+import bio.terra.profile.app.controller.UnauthenticatedApiController;
+import bio.terra.profile.db.ProfileDao;
+import bio.terra.profile.service.crl.AzureCrlService;
+import bio.terra.profile.service.crl.GcpCrlService;
+import bio.terra.profile.service.iam.SamService;
+import bio.terra.profile.service.iam.model.SamResourceType;
+import bio.terra.profile.service.job.JobService;
+import bio.terra.profile.service.profile.exception.ProfileNotFoundException;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import bio.terra.profile.service.status.ProfileStatusService;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.jdbc.JdbcRepositoriesAutoConfiguration;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
+
+@Tag("provider-test")
+@Provider("bpm-provider")
+@PactBroker
+// for local testing, put any test pacts in the service/pacts folder.
+// then comment out the above line, and uncomment the following line
+// @PactFolder("pacts")
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = Main.class,
+    properties = {
+      "profile.profile-database.initialize-on-start=false",
+      "profile.profile-database.upgrade-on-start=false"
+    })
+@EnableAutoConfiguration(
+    exclude = {DataSourceAutoConfiguration.class, JdbcRepositoriesAutoConfiguration.class})
+@ComponentScan(
+    basePackages = {"bio.terra.profile.app.controller", "bio.terra.profile.service"},
+    excludeFilters = {
+      @ComponentScan.Filter(
+          type = FilterType.ASSIGNABLE_TYPE,
+          value = {BeanConfig.class, ProfileDao.class}),
+      @ComponentScan.Filter(
+          type = FilterType.REGEX,
+          pattern = {"jdbcDialect", "jdbcCustomConversions", "jdbcMappingContext"})
+    })
+@ConfigurationPropertiesScan(basePackages = {"bio.terra.profile"})
```

Comment:
There's a lot of annotations here and seemingly a lot more than we need for other tests, could you give a little information about what these all do and why they're needed?

Reply:
There are some of these that, going back though and testing them out, are no longer necessary, because of some iteration on the spring configuration. That would be the `@ComponentScan` and `@ConfigurationPropertiesScan`. I'll remove those.

`@EnableAutoConfiguration` is actually already present implicitly on other tests, it's just coming by way in inheriting from `BaseSpringUnitTest`, which is getting its configuration from `Main`. This is just adding an additional auto-config class to ignore.

The remainder are for pact